# _Pseudomonas_ pangenome
This repository contains the codes used for generating the data contained in the article "Comparative genomics on the genus Pseudomonas reveals metabolic functions and thousands of genes associated to different environments" (Saati-Santamaría, Z., Baroncelli, R., Rivas, R., García-Fraile, P.).

<p align="center">
  <img width="700" alt="Captura de pantalla 2022-01-09 a las 14 40 51" src="https://user-images.githubusercontent.com/50806485/148684683-ac4fd96c-f493-4be9-9ee5-5e48caa6039a.png">


---
## Guide

- To inspect the codes and programs used to annotate genomes or proteomes, and to evaluate genome completeness and quality go to the [genomes_annotation](./analyses/genomes_annotation.md) section. This section details many diverse annotation steps.

- To see how we created the phylogenetic tree of _Pseudomonas_ , go to [phylogeny](./analyses/phylogeny.md).

- To see how we built an analyze the _Pseudomonas_ pangenome, and then searched for proteins or functions associated with the _Pseudomonas_ isolation sources go to [pangenome_creation_and_enrichment_analyses](./analyses/pangenome_creation_and_enrichment_analyses.md).

- To inspect the codes used for the generation of graphics and figures, go to the [Figures](./Figures/figures.md) section. This folder also includes the [Source Data](./Figures/Source_data) needed to create each graphic.

---

### Citation
Saati-Santamaría, Z., Baroncelli, R., Rivas, R., García-Fraile, P. Comparative genomics on the genus _Pseudomonas_ reveals metabolic functions and thousands of genes associated to different environments. Nature Ecology and Evolution. Under revision.

---

### Contact

If you have some questions related to the workflow of this work, please, write to [Zaki Saati Santamaría](https://www.researchgate.net/profile/Zaki-Saati-Santamaria)  (zakisaati@usal.es)

----
