# _Pseudomonas_ pangenome
This repository contains the codes used for generating the data contained in the article "Comparative genomics of the genus _Pseudomonas_ reveals host- and environment-specific evolution".

<p align="center">
 
 <img width="861" alt="Captura de pantalla 2022-09-15 a las 9 53 30" src="https://user-images.githubusercontent.com/50806485/190356913-2fe51558-4c36-462b-b286-7c154a29f4ef.png">



---
## Guide
- The [Analyses](./analyses/) folder contains data and codes needed to compute the genome analyses.
  
  - To inspect the codes and programs used to annotate genomes or proteomes, and to evaluate genome completeness and quality go to the [genomes_annotation](./analyses/genomes_annotation.md) section. This section details many diverse annotation steps.

  - To see how we created the phylogenetic tree of _Pseudomonas_ and classify the species, go to [phylogeny](./analyses/phylogeny.md).

  - To see how we built an analyze the _Pseudomonas_ pangenome, and then searched for proteins or functions associated with the _Pseudomonas_ isolation sources go to [pangenome_creation_and_enrichment_analyses](./analyses/pangenome_creation_and_enrichment_analyses.md). 

- To inspect the codes used for the generation of graphics and figures, go to the [Figures](./Figures/figures.md) section. This folder also includes the [Source Data](./Figures/Source_data) needed to create each graphic.

---

### Citation
Saati-Santamaría, Z., Baroncelli, R., Rivas, R., García-Fraile, P. Comparative genomics of the genus _Pseudomonas_ reveals host- and environment-specific evolution. Under revision.

---

### Contact

If you have some questions related to the workflow of this work, please, write to zakisaati@usal.es
  
----
